content/doc/developer/architecture/web.adoc

```diff
@@ -0,0 +1,23 @@
+---
+title: Web Framework
+layout: developersection
+summary: How Jenkins uses the Stapler framework
+---
+
+== Request Handling
+
+Jenkins classes are bound to URLs by using http://stapler.kohsuke.org[Stapler]. The singleton +jenkinsdoc:Jenkins[]+ instance is bound to the context root (i.e. "/") URL, and the rest of the objects are bound according to their reachability from this root object. Stapler uses reflection to recursively determine how to process any given URL. A few examples of how the URL `/foo/bar` could be processed:
+
+* A `getFoo(String)` is defined on the `Jenkins` object, and Stapler passes `bar` as a parameter. The object returned has a method called `doIndex(…)` that gets called and renders the response.
+* `getFoo()` or `doFoo()` are defined. They return an object that has a `getBar` or `doBar` method. The object returned from that has an associated `index.jelly` or `index.groovy` view.
```

Comment:
No, `doFoo()` would be invoked as a web method, and it would have to use `StaplerRequest.getRestOfPath()` to do any further processing if it cared about `/bar`. Similarly `doBar` would be invoked as a web method—not returning an object for further binding.